packages/plugin-styled-components/src/index.ts

```diff
@@ -0,0 +1,79 @@
+import {
+  DefaultRsbuildPlugin,
+  getDefaultStyledComponentsConfig,
+  mergeChainedOptions,
+  ChainedConfig,
+} from '@rsbuild/shared';
+
+/**
+ * the options of [babel-plugin-styled-components](https://github.com/styled-components/babel-plugin-styled-components) or [rspackExperiments.styledComponents](https://www.rspack.dev/guide/loader.html#optionsrspackexperimentsstyledcomponents).
+ */
+type StyledComponentsOptions = {
+  displayName?: boolean;
+  ssr?: boolean;
+  fileName?: boolean;
+  meaninglessFileNames?: string[];
+  namespace?: string;
+  topLevelImportPaths?: string[];
+  transpileTemplateLiterals?: boolean;
+  minify?: boolean;
+  pure?: boolean;
+  cssProps?: boolean;
+};
+
+export const pluginStyledComponents = (
+  userConfig: ChainedConfig<StyledComponentsOptions> = {},
+): DefaultRsbuildPlugin => ({
+  name: 'plugin-styled-components',
+
+  setup(api) {
+    api.modifyBundlerChain(async (chain, { CHAIN_ID, isProd, target }) => {
+      const { bundlerType } = api.context;
+
+      const isSSR = target === 'node';
```

Comment:
If targets contains node, I think both client and server should enable the `ssr` option to keep classname consistent. 

Please check the document of babel-plugin-styled-components.